internal/telemetry/scc.go

```diff
@@ -0,0 +1,48 @@
+package telemetry
+
+import (
+	"github.com/SUSE/connect-ng/pkg/registration"
+	"github.com/rancher/scc-operator/internal/rancher"
+)
+
+type subscriptionInfo struct {
+	rancherUUID string
+	product     string
+	version     string
+	arch        string
+	git         string
+}
+
+type MetricsWrapper struct {
+	Data             map[string]any
+	subscriptionInfo subscriptionInfo
+}
+
+func NewMetricsWrapper(data map[string]any) MetricsWrapper {
+	var subInfo subscriptionInfo
+	subscriptionData := data["subscription"].(map[string]interface{})
```

Comment:
Is there no chance that this map/secret does not contain all the data we need and causes this to panic?

Reply:
From my understanding of how we have the Telemetry code in `r/r` work that shouldn't be very likely. Potentially possible, so I can't say it's impossible. However, the intended contract between `r/r` telemetry and the `scc-operator` (_or any other potential future consumer of_ `SecretRequest`) is intended to be: "If the metric secret exists it's valid metrics".

Reply:
Sounds good, I can definitely do that in a follow up PR